core-processor/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -435,7 +456,10 @@ public class BeanDefinitionWriter extends AbstractClassFileWriter implements Bea
             AbstractInitializableBeanDefinition.AnnotationReference[].class, // annotationInjection
             ExecutableMethodsDefinition.class, // executableMethodsDefinition
             Map.class, // typeArgumentsMap
-            AbstractInitializableBeanDefinition.PrecalculatedInfo.class // precalculated info
+            AbstractInitializableBeanDefinition.PrecalculatedInfo.class, // precalculated info
+            Condition[].class, // pre conditions
+            Condition[].class, // post conditions
+            Throwable.class // failed initialization
```

Comment:
can you change this slightly to try lookup this constructor otherwise if not there fallback to the other constructor. Otherwise we will hit binary compatibility issues for all the modules again.

Reply:
This will not break anything, that's the new super constructor. We have some modules that have been compiled with previous version and tests are passing.

Reply:
yes but if there is an old version of `micronaut-inject` then the lookup will fail which means we will have to release new versions of every minor release

Reply:
How is it possible to have an older version of micronaut-inject? Inject and core-processor should always come with the same version.

Reply:
see https://github.com/micronaut-projects/micronaut-platform/issues/1075